src/features/notes/components/AddNoteModal.js

```diff
@@ -75,17 +75,30 @@ const validationSchema = Yup.object().shape({
   title: Yup.string().required("Title is required!"),
 });
 
-const AddNoteModal = ({ isOpen, setIsOpen }) => {
+const AddNoteModal = ({
+  isOpen,
+  setIsOpen,
+  initialValues,
+  isEdit = false,
+  noteId,
+  noteCreator,
+  activeButtonText = "Add Note",
+  cancelButtonText = "Cancle",
+}) => {
+  const emptyState = {};
   const dispatch = useDispatch();
+  const myId = useSelector((state) => state.auth.me.id);
+  const disabled = isEdit && myId !== noteCreator;
   const formik = useFormik({
-    initialValues: {
-      title: "",
-      body: "",
-      isPrivate: 1,
-    },
+    initialValues: initialValues || emptyState,
     onSubmit: (values, { resetForm }) => {
-      dispatch(addNote(values));
-      resetForm({});
+      if (isEdit) {
```

Comment:
- Prevent submit if the user can't edit the note.

Reply:
Added if condition to check that before dispatching an action

Reply:
Good job.